linkerd/protocol/h2/src/main/scala/com/twitter/finagle/buoyant/h2/LinkerdHeaders.scala

```diff
@@ -435,13 +435,31 @@ object LinkerdHeaders {
   }
 
   class ClearMiscServerFilter extends SimpleFilter[Request, Response] {
-    def apply(req: Request, service: Service[Request, Response]) = {
-      for ((k, _) <- req.headers.toSeq) {
+
+    private[this] def clearLinkerdHeaders(headers: Headers) = {
+      for ((k, _) <- headers.toSeq) {
         if (k.toLowerCase.startsWith(LinkerdHeaders.Prefix)) {
-          req.headers.remove(k)
+          headers.remove(k)
+        }
+      }
+    }
+
+    def apply(req: Request, service: Service[Request, Response]) = {
+      clearLinkerdHeaders(req.headers)
+      service(req).map { resp =>
+        val rsp = resp.dup()
```

Comment:
I notice that we never use `rsp` except to hold its headers.  This is basically equivalent but it might be a bit more clear:
```
val headers = resp.headers.dup()
...
clearLinkerdHeaders(headers)
Response(headers, Stream.empty())
```

Reply:
Yea, that is much more straightforward. Thanks for the suggestion!